callstats-java-sdk/src/main/java/io/callstats/sdk/CallStats.java

```diff
@@ -0,0 +1,114 @@
+		if(appId <= 0 || appSecret == null || bridgeId == null || endpointInfo == null) {
+			logger.error("intialize: Arguments cannot be null");
+			throw new IllegalArgumentException("intialize: Arguments cannot be null");
+		}
+		
+		this.appId = appId;
+		this.appSecret = appSecret;
+		this.bridgeId = bridgeId;
+		this.listener = callStatsInitListener;
+		this.endpointInfo = endpointInfo;
+		
+		httpClient = new CallStatsHttpClient();
+		authenticator = new CallStatsAuthenticator(new CallStatsInitListener() {
+			
+			public void onInitialized(String msg) {
+				listener.onInitialized(msg);
+				setInitialized(true);
+				CallStatsBridgeKeepAliveManager bridgeKeepAliveManager = new CallStatsBridgeKeepAliveManager(appId, bridgeId, authenticator.getToken(), httpClient);
+				bridgeKeepAliveManager.startKeepAliveSender();
+			}
+			
+			public void onError(CallStatsErrors error, String errMsg) {
+				listener.onError(error, errMsg);;				
+			}
+		});		
+		authenticator.doAuthentication(appId, this.appSecret, bridgeId, httpClient);
+		
+//		CallStatsAsyncHttpClient httpClient1;
+//		httpClient1 = new CallStatsAsyncHttpClient();	
+//		authenticator.doAuthentication1(appId, appSecret, bridgeId,httpClient1);
+	}
+	
+	
+	public void sendCallStatsBridgeEvent(BridgeStatusInfo bridgeStatusInfo) {	
+		if(isInitialized())	{			 
+			long epoch = System.currentTimeMillis()/1000;				
+			BridgeEventMessage eventMessage = new BridgeEventMessage(appId, bridgeId, CallStatsConst.CS_VERSION, CallStatsConst.END_POINT_TYPE, ""+epoch, authenticator.getToken(), bridgeStatusInfo, endpointInfo);
+			String requestMessageString = gson.toJson(eventMessage);
+			httpClient.sendAsyncHttpRequest(CallStatsConst.bridgeEventUrl,CallStatsConst.httpPostMethod, requestMessageString,new CallStatsHttpResponseListener() {
+				public void onResponse(HttpResponse response) {
+					
+					int responseStatus = response.getStatusLine().getStatusCode();
+					logger.debug("Response "+response.toString()+":"+responseStatus);
+					BridgeEventResponse eventResponseMessage;
+					try {
+						String responseString = EntityUtils.toString(response.getEntity());
+						eventResponseMessage  = gson.fromJson(responseString,BridgeEventResponse.class);	
+					} catch (ParseException e) {						
+						logger.error("ParseException "+e.getMessage()+":"+e.getStackTrace());
+						throw new RuntimeException(e);
+					} catch (IOException e) {
+						logger.error("IO Execption "+e.getMessage()+":"+e.getStackTrace());
+						throw new RuntimeException(e);					
+					}
+					if(responseStatus == 200) {
+						logger.debug("Response status is "+eventResponseMessage.getStatus()+":"+eventResponseMessage.getReason());
+					}
+				}
+				
+				public void onFailure(Exception e) {
+					logger.error("Response exception"+e.toString());
+				}
+				
+			});	
+		} else {
+			// TODO retransmission queue
```

Comment:
For the things that have not been implemented yet, probably it's better to use something like:
```
throw new UnsupportedOperationException("something not implemented yet")
```